src/Microsoft.OData.Client/ODataRequestMessageWrapper.cs

```diff
@@ -377,12 +377,12 @@ internal void FireSendingRequest2(Descriptor descriptor)
                 // For now, we don't think this adds a lot of value exposing on the public DataServiceClientRequestMessage class
                 // In future, we can always add it if customers ask for this. Erring on the side of keeping the public
                 // class simple.
-                var httpWebRequestMessage = this.requestMessage as HttpWebRequestMessage;
-                if (httpWebRequestMessage != null)
+                var httpClientRequestMessage = this.requestMessage as HttpClientRequestMessage;

```

Comment:
Maybe add a case for HttpRequestTransportMode.HttpWebRequestMessage.

Reply:
Is there any need for that if we are deprecating the HttpWebRequestMessage class?

Reply:
I think till we remove it completely we should leave out the functionality so any client that chooses to upgrade without changing the implementation is not affected when firing events.

Reply:
+1 for backwards compatibility.

---
In reply to: [540839611](https://github.com/OData/odata.net/pull/1953#discussion_r540839611) [](ancestors = 540839611)